src/features/environments/components/EnvironmentDropdown.tsx

```diff
@@ -100,13 +75,20 @@ export const EnvironmentDropdown = ({
             }
           >
             <StyledIconButton
+              component={Link}
+              to={`/${namespace}/new-environment`}
               style={{
                 textTransform: "none",
                 paddingRight: "0.4rem"
               }}
-              onClick={e =>
-                canCreate && onCreateNewEnvironmentTab(e, namespace)
-              }
+              onClick={event => {
+                if (!canCreate) {
```

Comment:
If `canCreate` is false this button is essentially disabled so call both event.preventDefault() and event.stopPropagation(); otherwise, we need to check if the namespace is already expanded, and if it is, we stop the click from propagating and causing the accordion component to collapse.